src/main.rs

```diff
@@ -1,4 +1,63 @@
 mod config;
 mod revaultd;
 
-fn main() {}
+use std::path::PathBuf;
+use std::{env, process};
+
+use config::{parse_config, Config};
+use revaultd::RevaultD;
+
+use daemonize_simple::Daemonize;
+
+fn parse_args(args: Vec<String>) -> Option<PathBuf> {
+    if args.len() == 1 {
+        return None;
+    }
+
+    if args.len() != 3 {
+        eprintln!("Unknown arguments '{:?}'.", args);
+        eprintln!("Only '--conf <configuration file path>' is supported.");
+        process::exit(1);
+    }
+
+    Some(PathBuf::from(args[2].to_owned()))
+}
+
+fn daemon_main(revaultd: RevaultD) {}
+
+fn main() {
+    let args = env::args().collect();
+    let conf_file = parse_args(args);
+
+    match parse_config(conf_file) {
```

Comment:
for less indentation, I think you can do:
```suggestion
    let config = parse_config(conf_file).unwrap_or_else(|e| {
           eprintln!("Error parsing config: {}", e);
            process::exit(1);
    });
```

Reply:
Nice, thanks